feat(types): add local structs for assets

GoodluckH opened the pull request.

GoodluckH committed:
feat(types): add local structs for assets

amsmith-pro commented on server/src/structs.rs:
```diff
@@ -0,0 +1,231 @@
+    pub name: Option<String>,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
```
I think maybe groups can be delegated but not individual assets

A group of asset size one can be delegated

GoodluckH replied:
[21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)

amsmith-pro commented on server/src/structs.rs:
```diff
@@ -0,0 +1,231 @@
+    /// If the vector is empty, the aircraft is available to everyone.
+    /// Otherwise, the aircraft is only available to the clients in the
+    /// vector.
+    pub restricted_to: Vec<Uuid>,
```
is `whitelist` fine?

noted these are temporary

GoodluckH replied:
[21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)

amsmith-pro commented on server/src/structs.rs:
```diff
@@ -0,0 +1,231 @@
+    pub name: String,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
```
Same comment about groups being delegated, not individual assets

GoodluckH replied:
[21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)

GoodluckH committed:
refactor(assets): add structs and unit tests

GoodluckH commented:
Squashed those commits into one to fix some failed github actions.

Refactored the code to have a Basics struct that's shared by all asset types -- reduces repetitive codes.

Also wrote unit tests.
